src/pool/srv_cli.c

```diff
@@ -491,3 +497,615 @@ dsc_pool_svc_query(uuid_t pool_uuid, d_rank_list_t *ps_ranks, uint64_t deadline,
+			  d_rank_t rank, uint32_t tgt_idx, daos_target_info_t *ti)
+{
+	struct pool_query_target_arg arg = {
+		.pqta_rank	= rank,
+		.pqta_tgt_idx	= tgt_idx,
+		.pqta_info	= ti
+	};
+
+	if (ti == NULL)
+		return -DER_INVAL;
+	D_DEBUG(DB_MGMT, DF_UUID ": Querying pool target %u\n", DP_UUID(pool_uuid), tgt_idx);
+	return dsc_pool_svc_call(pool_uuid, ps_ranks, &pool_query_target_cbs, &arg, deadline);
+}
+
+struct pool_evict_arg {
+	uuid_t   *pea_handles;
+	size_t    pea_n_handles;
+	char     *pea_machine;
+	uint32_t  pea_destroy;
+	uint32_t  pea_force;
+	uint32_t *pea_count;
+};
+
+static int
+pool_evict_init(uuid_t pool_uuid, crt_rpc_t *rpc, void *varg)
+{
+	struct pool_evict_arg *arg = varg;
+	struct pool_evict_in  *in  = crt_req_get(rpc);
+
+	in->pvi_hdls.ca_arrays = arg->pea_handles;
+	in->pvi_hdls.ca_count  = arg->pea_n_handles;
+	in->pvi_machine        = arg->pea_machine;
+	/* Pool destroy (force=false): assert no open handles / do not evict.
+	 * Pool destroy (force=true): evict any/all open handles on the pool.
+	 */
+	in->pvi_pool_destroy       = arg->pea_destroy;
+	in->pvi_pool_destroy_force = arg->pea_force;
+	return 0;
+}
+
+static int
+pool_evict_consume(uuid_t pool_uuid, crt_rpc_t *rpc, void *varg)
+{
+	struct pool_evict_arg *arg = varg;
+	struct pool_evict_out *out = crt_reply_get(rpc);
+	int                    rc  = out->pvo_op.po_rc;
+
+	if (rc != 0)
+		DL_ERROR(rc, DF_UUID ": pool destroy failed to evict handles", DP_UUID(pool_uuid));
+	if (arg->pea_count != NULL)
+		*arg->pea_count = out->pvo_n_hdls_evicted;
+	return rc;
+}
+
+static struct dsc_pool_svc_call_cbs pool_evict_cbs = {
+	.pscc_op	= POOL_EVICT,
+	.pscc_init	= pool_evict_init,
+	.pscc_consume	= pool_evict_consume,
+	.pscc_fini	= NULL
+};
+
+/**
+ * Test and (if applicable based on destroy and force option) evict all open
+ * handles on a pool.
+ *
+ * \param[in]	pool_uuid	UUID of the pool
+ * \param[in]	ranks		Pool service replicas
+ * \param[in]	deadline	Unix time deadline in milliseconds
+ * \param[in]	handles		List of handles to selectively evict
+ * \param[in]	n_handles	Number of items in handles
+ * \param[in]	destroy		If true the evict request is a destroy request
+ * \param[in]	force		If true and destroy is true request all handles
+ *				be forcibly evicted
+ * \param[in]   machine		Hostname to use as filter for evicting handles
+ * \param[out]	count		Number of handles evicted
+ *
+ * \return	0		Success
+ *		-DER_BUSY	Open pool handles exist and no force requested
+ */
+int
+dsc_pool_svc_check_evict(uuid_t pool_uuid, d_rank_list_t *ranks, uint64_t deadline, uuid_t *handles,
+			 size_t n_handles, uint32_t destroy, uint32_t force, char *machine,
+			 uint32_t *count)
+{
+	struct pool_evict_arg arg = {
+		.pea_handles	= handles,
+		.pea_n_handles	= n_handles,
+		.pea_machine	= machine,
+		.pea_destroy	= destroy,
+		.pea_force	= force,
+		.pea_count	= count
+	};
+
+	D_DEBUG(DB_MGMT, DF_UUID ": Destroy pool (force: %d), inspect/evict handles\n",
```

Comment:
log message may want to not always say "Destroy", depending on the uint32_t destroy argument setting.